contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -4014,93 +4014,56 @@ GRANT EXECUTE ON FUNCTION sys.fn_listextendedproperty TO PUBLIC;
 -- Matches and returns column length of the corresponding column of the given table
 CREATE OR REPLACE FUNCTION sys.COL_LENGTH(IN object_name TEXT, IN column_name TEXT)
 RETURNS SMALLINT AS $BODY$
-    DECLARE
-        col_name TEXT;
-        object_id oid;
-        column_id INT;
-        column_length INT;
-        column_data_type TEXT;
-        column_precision INT;
-    BEGIN
-        -- Get the object ID for the provided object_name
-        object_id = sys.OBJECT_ID(object_name);
-        IF object_id IS NULL THEN
-            RETURN NULL;
-        END IF;
+DECLARE
+    col_name TEXT;
+    object_id oid;
+    column_id INT;
+    column_length INT;
+    column_data_type TEXT;
+    typelen INT;
+    typemod INT;
+BEGIN
+    -- Get the object ID for the provided object_name
+    object_id := sys.OBJECT_ID(object_name);
```

Comment:
Here we are only interested in tables, so its better to provide object_type as input in object_id function to avoid unnecessary lookups.